scripts/get_pulls_in_range.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+
+import argparse
+import getpass
+import netrc
+import os
+import sys
+
+import pygit2
+from github import Github
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        epilog='''Get a list of pull requests associated
+        with a range of commits in a GitHub repository.
+
+        For prompt-less operation, either set up a ~/.netrc with
+        credentials for github.com, or define the GITHUB_USERNAME and
+        GITHUB_TOKEN environment variables.''')
+    parser.add_argument('gh_repo',
+                        help='''repository in <organization>/<repo> format,
+                        like nrfconnect/sdk-nrf''')
+    parser.add_argument('local_path',
+                        help='path to repository on the file system')
+    parser.add_argument('start', help='SHA of start commit in the range')
+    parser.add_argument('end', help='SHA of end commit in the range')
+    return parser.parse_args()
+
+def get_gh_credentials():
+    user = None
+    token = None
+
+    try:
+        nrc = netrc.netrc()
+    except FileNotFoundError:
+        nrc = None
+    except netrc.NetrcParseError:
+        print('parse error in netrc file, falling back on environment',
+              file=sys.stderr)
+        nrc = None
+
+    if nrc is not None:
+        auth = nrc.authenticators('github.com')
+        if auth is not None:
+            login, _, password = auth
+            user = login
+            token = password
+
+    if user is None:
+        user = os.environ.get('GITHUB_USERNAME')
```

Comment:
You don't need a username if you are using a token

Reply:
Should be fixed now.